File: src-tauri/src/widgets/feed.rs
//! 인앱 알림 피드 (이벤트/결과 통합)

use serde::Serialize;
use tauri::{AppHandle, Emitter, Manager};

#[derive(Clone, Serialize)]
pub struct FeedItem {
    pub ts: i64,
    pub icon: String,
    pub title: String,
    pub body: String,
}

const MAX_ITEMS: usize = 60;

pub fn list(st: &crate::app::AppState) -> Vec<FeedItem> {
    st.feed.lock().clone()
}

pub fn push(app: &AppHandle, icon: &str, title: &str, body: &str) {
    let item = FeedItem {
        ts: crate::platform::now_ms(),
        icon: icon.into(),
        title: title.into(),
        body: body.into(),
    };
    {
        let st = app.state::<crate::app::AppState>();
        let mut feed = st.feed.lock();
        feed.insert(0, item.clone());
        feed.truncate(MAX_ITEMS);
    }
    let _ = app.emit("feed-new", &item);
}

File: goport/feed.go
package main

// 인앱 알림 피드

type FeedItem struct {
	Ts    int64  `json:"ts"`
	Icon  string `json:"icon"`
	Title string `json:"title"`
	Body  string `json:"body"`
}

const maxFeedItems = 60

func feedList(st *AppState) []FeedItem {
	st.feedMu.Lock()
	defer st.feedMu.Unlock()
	out := make([]FeedItem, len(st.feed))
	copy(out, st.feed)
	return out
}

func feedPush(st *AppState, icon, title, body string) {
	item := FeedItem{Ts: nowMs(), Icon: icon, Title: title, Body: body}
	st.feedMu.Lock()
	st.feed = append([]FeedItem{item}, st.feed...)
	if len(st.feed) > maxFeedItems {
		st.feed = st.feed[:maxFeedItems]
	}
	st.feedMu.Unlock()
	st.events.Publish("feed-new", item)
}

func feedClear(st *AppState) []FeedItem {
	st.feedMu.Lock()
	st.feed = []FeedItem{}
	st.feedMu.Unlock()
	st.events.Publish("feed", []FeedItem{})
	return []FeedItem{}
}
